ncbi/src/main/java/org/reactome/release/NCBIEntry.java

```diff
@@ -0,0 +1,201 @@
+			ncbiEntries.add(
+				new NCBIEntry(uniprot, uniprotToNCBIGene.get(uniprot))
+			);
+		}
+		Collections.sort(ncbiEntries);
+
+		logger.info("Finished generating UniProt accession to NCBI Gene mapping");
+
+		return ncbiEntries;
+	}
+
+	/**
+	 * Compares UniProt accession values of this object and parameter
+	 * @param o NCBIEntry object to compare
+	 * @return Value of String compare between this UniProt accession and the parameter's UniProt accession
+	 */
+	@Override
+	public int compareTo(@Nonnull NCBIEntry o) {
+		return this.getUniprotAccession().compareTo(o.getUniprotAccession());
+	}
+
+	/**
+	 * Checks equality based on object type and value of UniProt accession, UniProt display name, and NCBI Gene ids
+	 * @param o Object to check for equality with the calling NCBI Entry object.
+	 * @return <code>true</code> if the same object or an NCBI Entry object with the same UniProt accession,
+	 * UniProt display name, and NCBI Gene ids.  Returns <code>false</code> otherwise.
+	 */
+	@Override
+	public boolean equals(Object o) {
+		if (o == this) {
+			return true;
+		}
+
+		if (!(o instanceof NCBIEntry)) {
+			return false;
+		}
+
+		NCBIEntry other = (NCBIEntry) o;
+
+		return other.getUniprotAccession().equals(this.getUniprotAccession()) &&
+			   other.getUniprotDisplayName().equals(this.getUniprotDisplayName()) &&
+			   other.getNcbiGeneIds().equals(this.getNcbiGeneIds());
+	}
+
+	/**
+	 * Retrieves a hash code based on the object's set fields
+	 * @return Hash code of NCBI Entry object
+	 */
+	@Override
+	public int hashCode() {
+		return Objects.hash(getUniprotAccession(), getUniprotDisplayName(), getNcbiGeneIds());
+	}
+
+	/**
+	 * Retrieves a String representation of the defining data of the NCBI Entry object
+	 * @return String representation of NCBI Entry object
+	 */
+	@Override
+	public String toString() {
+		return this.getUniprotDisplayName() + " with NCBI Gene ids " + this.getNcbiGeneIds();
+	}
+
+	/**
+	 * Returns the XML String describing the relationship between the object's UniProt entry and an NCBI Gene
+	 * @param ncbiGene NCBI Gene identifier
+	 * @return XML String for NCBI Gene Entity Link
+	 */
+	public String getEntityLinkXML(String ncbiGene) {
+		return getLinkXML(
+			ncbiGene,
+			"&entity.base.url;",
+			getUniprotAccession(),
+			"Reactome Entity:" + getUniprotAccession()
+		);
+	}
+
+	/**
+	 * Returns the XML String describing the relationship between a Reactome pathway and an NCBI Gene
+	 * @param ncbiGene NCBI Gene identifier
+	 * @param pathway Reactome Event representing a pathway
+	 * @return XML String for NCBI Gene Entity Link
+	 */
+	public String getEventLinkXML(String ncbiGene, ReactomeEvent pathway) {
+		return getLinkXML(
+			ncbiGene,
+			"&event.base.url;",
+			pathway.getStableIdentifier(),
+			"Reactome Event:" + pathway.getName()
+		);
+	}
+
+	private String getLinkXML(String ncbiGene, String base, String rule, String urlName) {
+		return String.join(System.lineSeparator(),
+			"\t<Link>",
```

Comment:
You used an 'indent' method up above for a similar problem as here. That looked a bit cleaner and would make the code more consistent

Reply:
Yes, you're right.  I actually think I'm going to try to generate the XML in a way that's more commonly done in Java.  I'll ask you guys to take a look after I've made those changes.